src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -362,8 +375,8 @@ export function findPreferredKernel(
                     !notebookMetadata && // If we don't have metadata, only then should we compare against the interpreter.
                     isKernelRegisteredByUs(spec) === 'newVersion' &&
                     isPythonKernelSpec(spec) &&
-                    interpreter &&
-                    spec.name === getInterpreterKernelSpecName(interpreter)
+                    preferredInterpreter &&
+                    spec.name === getInterpreterKernelSpecName(preferredInterpreter)
                 ) {
                     // This is a perfect match.
                     score += 100;
```

Comment:
I believe this path was supposed to do the same thing. How come it doesn't go down this path?

Reply:
No never mind. Spec.name wouldn't match.